scanpipe/tests/test_pipes.py

```diff
@@ -700,6 +701,43 @@ def test_scanpipe_pipes_rootfs_has_hash_diff(self):
         codebase_resource = CodebaseResource(sha256="sha256", md5="md5")
         self.assertFalse(rootfs.has_hash_diff(install_file, codebase_resource))
 
+    def test_scanpipe_pipes_glc_scan_file_for_license(self):
+        input_location = str(self.data_location / "notice.NOTICE")
+        scan_results = glc.scan_file_for_license(input_location)
+        expected = [
+            "path",
+            "licenses",
+            "license_expressions",
+            "copyrights",
+            "scan_errors",
+        ]
+        self.assertEqual(sorted(expected), sorted(scan_results.keys()))
+
+    def test_scanpipe_pipes_glc_update_codebase_resources(self):
+        project1 = Project.objects.create(name="Analysis")
+        codebase_resource1 = CodebaseResource.objects.create(
+            project=project1, path="not available"
+        )
+        self.assertEqual(0, project1.projecterrors.count())
+
+        scan_results = glc.scan_file_for_license(codebase_resource1.location)
+        glc.update_codebase_resources(project1, [scan_results])
+
+        codebase_resource1.refresh_from_db()
+        self.assertEqual("scanned-with-error", codebase_resource1.status)
+        self.assertEqual(1, project1.projecterrors.count())
+
+        copy_inputs([self.data_location / "notice.NOTICE"], project1.codebase_path)
+        codebase_resource2 = CodebaseResource.objects.create(
+            project=project1, path="notice.NOTICE"
+        )
+
+        scan_results = glc.scan_file_for_license(codebase_resource2.location)
+        glc.update_codebase_resources(project1, [scan_results])
+
+        codebase_resource2.refresh_from_db()
+        self.assertEqual("scanned", codebase_resource2.status)
+
```

Comment:
We could add somme extra assertion on the values set on the `codebase_resource2`.
Maybe on `licenses`, `copyrights`, ... everything that have been updated by the update_codebase_resources.

Reply:
It was my original plan but licenseclassifier is not able to produce results for `notice.NOTICE`. If we can have some other file in data folder then it might help.

Reply:
@AvishrantsSh You're welcome to add test data that produce relevant results for your tests.